specification/resources/gen-ai/GenAIAPI_CancelKnowledgeBaseIndexingJobPublic.yml

```
@@ -0,0 +1,40 @@
+description: To cancel an indexing job for a knowledge base, send a PUT request to
+  `/v2/gen-ai/indexing_jobs/{uuid}/cancel`.
+operationId: Cancel_Indexing_Job_for_a_Knowledge_Base
+parameters:
+- description: A unique identifier for an indexing job.
+  example: '"123e4567-e89b-12d3-a456-426614174000"'
+  in: path
+  name: uuid
+  required: true
+  type: string
+requestBody:
+  content:
+    application/json:
+      schema:
+        $ref: ./definitions.yml#/apiCancelKnowledgeBaseIndexingJobInputPublic
+responses:
+  "200":
+    content:
+      application/json:
+        schema:
+          $ref: ./definitions.yml#/apiCancelKnowledgeBaseIndexingJobOutput
+    description: A successful response.
+    headers:
+      ratelimit-limit:
+        $ref: ../../shared/headers.yml#/ratelimit-limit
+      ratelimit-remaining:
+        $ref: ../../shared/headers.yml#/ratelimit-remaining
+      ratelimit-reset:
+        $ref: ../../shared/headers.yml#/ratelimit-reset
+  "401":
+    $ref: ../../shared/responses/unauthorized.yml
+  "500":
+    $ref: ../../shared/responses/server_error.yml
+  default:
+    $ref: ../../shared/responses/unexpected_error.yml
+summary: Cancel Indexing Job for a Knowledge Base
+tags:
+- GenAI Platform
+x-codeSamples:
+- $ref: examples/curl/GenAIAPI_CancelKnowledgeBaseIndexingJobPublic.yml
```

```suggestion
- $ref: examples/curl/GenAIAPI_CancelKnowledgeBaseIndexingJobPublic.yml
security:
  - bearer_auth:
    - 'genai:update'
```

We also need to document the token scopes required for each operation. This can be done by adding a `security` section with the appropriate scopes under `bearer_auth`.